### DISCUSSION TITLE
host volumes owned by systemd-resolve (instead of www-data)

### DISCUSSION THREAD

**Discussion by ThierryS75** [netalertx/NetAlertX#872] [Q&A] [answered]
Hi All,

After migrating from pialert (under which host volumes config and db were owned by www-data) and using the docker-compose.yml for NetAlertX, my config, db and logs volumes are owned by systemd-resolve. Is this normal?

Thank you in advance for any help,

Thierry

**Comment by jokob-sk** [comment]
Hi @ThierryS75 ,

does this guide help?

https://github.com/jokob-sk/NetAlertX/blob/main/docs/FILE_PERMISSIONS.md

**Comment by ThierryS75** [comment]
Hi @jokob-sk, 

Thank you - I had looked at this before posting, unfortunately it does not help:

- My previous pialert install had pialert.db* and pialert.conf* owned by www-data:www-data (uid 33, gid 33)
- My new NetalertX install changed ownership of the db, config and logs directories, as well as their contents, from pi:pi to systemd-resolve:82 (uid 102, gid:82) upon first start

[FILE_PERMISSIONS.md](https://github.com/jokob-sk/NetAlertX/blob/main/docs/FILE_PERMISSIONS.md) states that this should be nginx:www-data (101:82). On my system, uid 101 is for systemd-network (101:102). User systemd-resolve is 102:103.

Could it be that NetAlertX detected that uid 101 was already in use (on the host system, not in the container???), and decided to use uid 102 instead?

Kind regards,

Thierry

**Reply by jokob-sk** [reply]
Hi,

I'm not sure if that is possible. Here is the code that ensures the correct ownership of the folders in the container. 

https://github.com/jokob-sk/NetAlertX/blob/f426d7b96078bc49292b0fe22a76d584c2c8bd1d/dockerfiles/setup.sh#L113

@vladaurosh implemented the permission changes a while ago, not sure if he can chip in.

**Comment by vladaurosh** [comment]
Hi @ThierryS75 

Netalertx recently had a change in base docker image (it's using nginx:alpine docker image) and it's not possible anymore to set custom user id and group id. 
Base image has user nginx with id 101 and gid 82, hence db, config and logs directories will need to be owned by user with same id and gid. 
On different host operation systems id 101 and gid 82 can map to different users and groups (like in your case), or not to map to any existing user/group. 
And both these are fine.

Are you experiencing issues with this uid and gid?

**Reply by ThierryS75** [reply]
Hi @vladaurosh,

Thank you for clarifying - no issues I can see so far. I was a bit concerned that this could be a security issue somehow. Any real risk to operate in that configuration?

Kind regards,

Thierry

**Reply by vladaurosh** [reply, marked as answer]
Shouldn't be any risks. :)

**Reply by ThierryS75** [reply]
Thank you, reassuring I suppose :-)

Best regards,

Thierry
